### PR TITLE
ASA-4007

### PR DESCRIPTION
**Pull request by vishalhcl-5960**
contains changes of the implementation of "login required: Sequence file" option.

### CODE REVIEW

File: src/main/java/com/hcl/appscan/sdk/scanners/dynamic/DASTScan.java [hunk 1]
```diff
@@ -41,12 +41,32 @@ public void run() throws ScannerException, InvalidTargetException {
 
 		Map<String, String> params = getProperties();
 		params.put(STARTING_URL, target);
-		
+
+		String scanLoginType = params.remove(LOGIN_TYPE);
+
+		if (scanLoginType.equals("Manual")) {
```
Comment (mattmurp) [inline src/main/java/com/hcl/appscan/sdk/scanners/dynamic/DASTScan.java:47]: Is there guaranteed to be a property LOGIN_TYPE? Switching this check to the following would avoid a potential NPE if the LOGIN_TYPE property doesn't exist:
`if("Manual".equals(scanLoginType)) {`

File: src/main/java/com/hcl/appscan/sdk/scanners/ASoCScan.java [hunk 2]
```diff
@@ -24,6 +24,7 @@ public abstract class ASoCScan implements IScan, ScanConstants, Serializable {
 	
 	private String m_target;
 	private String m_scanId;
+	private String m_trafficFileId;
```
Comment (mattmurp) [inline src/main/java/com/hcl/appscan/sdk/scanners/ASoCScan.java:27]: Is this variable needed? I don't see it used anywhere. If it is required it only applies to DAST scans, so it should be on the DASTScan class, not the ASoCScan class.

**Review by mattmurp — CHANGES_REQUESTED** [review]

File: src/main/java/com/hcl/appscan/sdk/scanners/ASoCScan.java (hunk 2 above)
Comment (vishalhcl) [reply src/main/java/com/hcl/appscan/sdk/scanners/ASoCScan.java]: Yeah Matt, We are not using this anywhere.

File: src/main/java/com/hcl/appscan/sdk/scanners/dynamic/DASTScan.java (hunk 1 above)
Comment (vishalhcl) [reply src/main/java/com/hcl/appscan/sdk/scanners/dynamic/DASTScan.java]: Made the changes accordingly.

**Review by mattmurp — APPROVED** [review]

**Pull request merged**